src/generator.rs

```diff
@@ -25,43 +35,109 @@ impl Generator {
             frequency,
             phase: match self.oscillator_state {
                 OscillatorState::Playing { phase, .. } => phase,
+                OscillatorState::Decaying { .. } => 0.0,
                 OscillatorState::Muted => 0.0,
             },
         };
     }
 
     pub fn mute(&mut self) {
```

Comment:
I wonder if the name of this should change to something like "note_off" to indicate that it's not an immediate mute, but respects note decay.

Reply:
Done.